src/main/java/bio/terra/service/dataset/flight/datadelete/DataDeletionStep.java

```diff
@@ -0,0 +1,86 @@
+package bio.terra.service.dataset.flight.datadelete;
+
+import bio.terra.common.FlightUtils;
+import bio.terra.model.DataDeletionRequest;
+import bio.terra.model.DataDeletionTableModel;
+import bio.terra.model.DeleteResponseModel;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.DatasetTable;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+
+public class DataDeletionStep implements Step {
+
+    private final BigQueryPdao bigQueryPdao;
+    private final DatasetService datasetService;
+
+    private static Logger logger = LoggerFactory.getLogger(DataDeletionStep.class);
+
+    public DataDeletionStep(BigQueryPdao bigQueryPdao, DatasetService datasetService) {
+        this.bigQueryPdao = bigQueryPdao;
+        this.datasetService = datasetService;
+    }
+
+    private String suffix(FlightContext context) {
```

Comment:
This method is used in a few places, perhaps pull it out into utils?

Reply:
Good point, was wondering the same.